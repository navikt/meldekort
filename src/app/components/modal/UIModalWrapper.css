.uimodal {
  width: 40rem;
  min-height: 15rem;
  position: relative;
}
.uimodal__content {
  margin-top: 1rem;
}
.uimodal__actions {
  position: absolute;
  bottom: 1rem;
}
.uimodal__actions-knapp {
  margin-left: 1rem;
}
.uimodal .uimodal__actions-knapp {
  margin-top: 0.5rem;
}
.modal-overskrift h2 {
  margin-bottom: 1rem;
}
