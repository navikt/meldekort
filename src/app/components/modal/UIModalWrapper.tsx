import * as React from 'react';
import { IModal, ModalKnapp } from '../../types/ui';
import { RootState } from '../../store/configureStore';
import { selectModal } from '../../selectors/ui';
import { Dispatch } from 'redux';
import { UiActions } from '../../actions/ui';
import { connect } from 'react-redux';
import KnappBase from 'nav-frontend-knapper';
import Modal from 'nav-frontend-modal';

interface MapStateToProps {
  modal: IModal;
}

interface MapDispatchToProps {
  skjulModal: () => void;
}

type UIModalWrapperProps = MapStateToProps & MapDispatchToProps;

const UIModalWrapper: React.FunctionComponent<UIModalWrapperProps> = ({
  modal,
  skjulModal
}) => {
  Modal.setAppElement('#meldekort__root');
  return (
    <Modal
      className={'uimodal'}
      isOpen={modal.visModal}
      onRequestClose={() => {
        modal.onRequestClose ? modal.onRequestClose() : skjulModal();
      }}
      contentLabel={'ui-modal'}
      closeButton={true}
    >
      <div className={'uimodal__content'}>{modal.content()}</div>
      <div className={'uimodal__actions-knapp'}>
        {modal.knapper &&
          modal.knapper.map((knapp: ModalKnapp) => {
            return (
              <KnappBase
                key={knapp.label}
                className={'uimodal__actions-knapp'}
                type={knapp.type}
                onClick={() => knapp.action()}
              >
                {knapp.label}
              </KnappBase>
            );
          })}
      </div>
    </Modal>
  );
};

const mapStateToProps = (state: RootState): MapStateToProps => {
  return {
    modal: selectModal(state),
  };
};

const mapDispatchToProps = (dispatch: Dispatch): MapDispatchToProps => {
  return {
    skjulModal: () => dispatch(UiActions.skjulModal()),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(UIModalWrapper);
