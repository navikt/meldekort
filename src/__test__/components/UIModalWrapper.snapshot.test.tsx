import * as React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ProviderWrapper from '../testSetup/providerWrapper';
import UIModalWrapper from '../../app/components/modal/UIModalWrapper';

Enzyme.configure({ adapter: new Adapter() });

it('UIModalWrapper', () => {
  const tree = shallow(
    <ProviderWrapper>
      <div id="meldekort__root">
        <UIModalWrapper />
      </div>
    </ProviderWrapper>
  );
  expect(tree).toMatchSnapshot();
});
