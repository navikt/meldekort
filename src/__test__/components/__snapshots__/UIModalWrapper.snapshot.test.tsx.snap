// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UIModalWrapper 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(IntlProvider)
    defaultLocale="nb"
    locale="nb"
  >
    <div
      id="meldekort__root"
    >
      <Connect(UIModalWrapper) />
    </div>
  </Connect(IntlProvider)>
</Provider>
`;
